apache/hadoop-ozone
# HDDS-3173. Provide better default JVM options

## What changes were proposed in this pull request?

The GC pressure on Datanode is high because of the retry cache. I found crashes due to the long GC pauses. I started to use the following JVM parameters:

```
-server -XX:ParallelGCThreads=8 -XX:+UseConcMarkSweepGC -XX:CMSInitiatingOccupancyFraction=70 -XX:+UseCMSInitiatingOccupancyOnly
```
Which provide stable output.

It would be great to detect the current version and add these parameters, if required.

But there are two problems:

    * Different java versions support different flags
    *  There could be conflicting flags (eg. if the user defines to use G1 we shouldn't add any other default parameters).

 * Solution: right now JDK14 is not yet supported (deprecated CMS)
 * Solution: If any of the -XX parameters are defined, JVM parameters won't be added

## What is the link to the Apache JIRA

https://issues.apache.org/jira/browse/HDDS-3173

## How was this patch tested?

 * Start docker compose with and without adding `HADOOP_OPTS` flag to the `docker-config` file. 
 * Check the used JVM opts

Unit tests:

```
cd hadoop-ozone/dist/src/test/shell
bats gc_opts.bats
```
